activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -191,6 +191,19 @@ def pick(*keys)
   def compact_blank
     reject(&:blank?)
   end
+
+  # Returns a new +Array+ where the order has been set to that provided in the +series+, based on the +key+ of the
+  # objects in the original enumerable.
+  #
+  #   [ Person.find(5), Person.find(3), Person.find(1) ].in_order_of(:id, [ 1, 5, 3 ])
+  #   => [ Person.find(1), Person.find(5), Person.find(3) ]
+  #
+  # If the +series+ include keys that have no corresponding element in the Enumerable, these are ignored.
+  # If the Enumerable has additional elements that aren't named in the +series+, these are not included in the result.
+  def in_order_of(key, series)
+    indexed_by_key = index_by(&key)
+    series.map { |s| indexed_by_key[s] }.compact
```

Comment:
Could we use `filter_map` here for Ruby > 2.7 ?

Reply:
Ah yes. Rails 7 will move to Ruby 2.7 anyway, so good to just use that.